roles/postfix/tasks/main.yml

```diff
@@ -0,0 +1,84 @@
+---
+# tasks file for roles/postfix
+- name: Postfix | add mailname
+  ansible.builtin.copy:
+    dest: "/etc/mailname"
+    content: |
+      "{{ ansible_facts['hostname'] }}"
+    force: false
+    owner: root
+    group: root
+    mode: "0644"
+
+- name: Postfix | Install Postfix
+  ansible.builtin.apt:
+    name: postfix
+    state: present
+    update_cache: true
+
+- name: Postfix | create networks file
+  ansible.builtin.copy:
+    src: "mynetworks"
+    dest: "{{ postfix_allowed_relay_hosts }}"
+    owner: root
+    group: root
+    mode: '0644'
+  when: running_on_server
+
+- name: Postfix | Configure Postfix as relay host
+  ansible.builtin.template:
+    src: main_cf.j2
+    dest: "{{ postfix_main_cf }}"
+    owner: root
+    group: root
+    mode: "0644"
+
+- name: Postfix | Add Proofpoint authentication credentials
+  ansible.builtin.copy:
+    dest: /etc/postfix/sasl_passwd
+    content: |
+      [{{ postfix_smtp_relay_host }}]:{{ postfix_smtp_relay_port }} {{ postfix_smtp_relay_username }}:{{ postfix_smtp_relay_password }}
+    mode: '0644'
+  when: running_on_server
+  notify: restart postfix
+
+- name: Postfix | Secure the password file
+  ansible.builtin.file:
+    path: /etc/postfix/sasl_passwd
+    owner: root
+    group: root
+    mode: '0600'
+  when: running_on_server
+
+- name: Postfix | Generate Postfix hash map for authentication
+  ansible.builtin.command: postmap /etc/postfix/sasl_passwd
+  changed_when: false
+  when: running_on_server
+  notify: restart postfix
+
+- name: Postfix | Add Postfix hosts
+  ansible.builtin.lineinfile:
```

Comment:
This still relies on the right IPs being in the right variable at the right time, and it doesn't allow for tracking those IPs over time under source control. I would much rather see a template here, with a variable that shows changes to the relay hosts over time. That way, for example, when we add a new VM to the inventory, we can add its IP to the relay hosts list in the same PR. And same goes for removing VMs.